athena-elasticsearch/src/main/java/com/amazonaws/connectors/athena/elasticsearch/ElasticsearchMetadataHandler.java

```diff
@@ -0,0 +1,282 @@
+    protected ElasticsearchMetadataHandler(AWSGlue awsGlue,
+                                           EncryptionKeyFactory keyFactory,
+                                           AWSSecretsManager awsSecretsManager,
+                                           AmazonAthena athena,
+                                           String spillBucket,
+                                           String spillPrefix,
+                                           AwsRestHighLevelClientFactory clientFactory)
+    {
+        super(awsGlue, keyFactory, awsSecretsManager, athena, SOURCE_TYPE, spillBucket, spillPrefix);
+        this.awsGlue = awsGlue;
+        this.clientFactory = clientFactory;
+
+        ElasticsearchHelper.setDomainMapping();
+    }
+
+    /**
+     * Used to get the list of domains (aka databases) for the Elasticsearch service.
+     * @param allocator Tool for creating and managing Apache Arrow Blocks.
+     * @param request Provides details on who made the request and which Athena catalog they are querying.
+     * @return A ListSchemasResponse which primarily contains a Set<String> of schema names and a catalog name
+     * corresponding the Athena catalog that was queried.
+     */
+    @Override
+    public ListSchemasResponse doListSchemaNames(BlockAllocator allocator, ListSchemasRequest request)
+    {
+        logger.info("doListSchemaNames: enter - " + request);
+
+        if (ElasticsearchHelper.isAutoDiscoverEndpoint()) {
+            // Since adding/deleting domains in Amazon Elasticsearch Service doesn't re-initialize
+            // the connector, the domainMap needs to be refreshed each time for Amazon ES.
+            ElasticsearchHelper.setDomainMapping();
+        }
+
+        return new ListSchemasResponse(request.getCatalogName(), ElasticsearchHelper.getDomainList());
+    }
+
+    /**
+      * Used to get the list of indices contained in the specified domain.
+      * @param allocator Tool for creating and managing Apache Arrow Blocks.
+      * @param request Provides details on who made the request and which Athena catalog and database they are querying.
+      * @return A ListTablesResponse which primarily contains a List<TableName> enumerating the tables in this
+      * catalog, database tuple. It also contains the catalog name corresponding the Athena catalog that was queried.
+      */
+    @Override
+    public ListTablesResponse doListTables(BlockAllocator allocator, ListTablesRequest request)
+    {
+        logger.info("doListTables: enter - " + request);
+
+        List<TableName> indices = new ArrayList<>();
+        String endpoint = ElasticsearchHelper.getDomainEndpoint(request.getSchemaName());
+
+        if (!endpoint.isEmpty()) {
+            AwsRestHighLevelClient client = clientFactory.getClient(endpoint);
+            try {
+                for (String index : client.getAliases()) {
+                    // Add all Indices except for kibana.
+                    if (index.contains("kibana")) {
```

Comment:
why hardcoding kibana here?

Reply:
Kibana which is part of the Elasticsearch stack, uses an internal index to store information related to its operation. It is not a user created index, so it should be excluded when we retrieve the indices.